es-core/src/utils/FileSystemUtil.cpp

```diff
@@ -740,16 +740,17 @@ namespace Utils
 		{
 			const std::string path = getGenericPath(_path);
 
-			// regular files and executables but not setuid, setgid, shared text (mode 0755)
-			const mode_t  mask = S_IFREG | S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH;
+			// regular files and executables, but not setuid, setgid, shared text
+			const mode_t mask = S_IFREG;
+			const mode_t mask_exec = S_IXUSR | S_IXGRP | S_IXOTH;
 			struct stat64 info;
 
 			// check if stat64 succeeded
 			if(stat64(path.c_str(), &info) != 0)
 				return false;
 
-			// check for mask attributes only
-			return ((info.st_mode & mask) == info.st_mode);
```

Comment:
Wouldn't the old code be enough if just changing it to compare to mask rather than info.st_mode?

Reply:
Changing to `(info.st_mode & mask) == mask` in the current implementation would match 755 and "higher", but not "lower" like 451 or 700 or 011. So this PR suggests to check only for at least one exec-bit (hence the two checks: one for regular files one for at least one exec bit). Execution can still fail but then the it is the user's responsibility (eg. mounting the script folder `noexec`). Rationale to allow also "lower" permissions: Compiled binaries need only a `x`-bit, interpreted scripts need `rx`.